src/main/entities/PasswordGenerator.java

```diff
@@ -0,0 +1,63 @@
+package entities;
+
+public class PasswordGenerator {
+
+    public static String generateLCLetters(int len) {
+        StringBuilder lc = new StringBuilder();
+        for (int i = 1; i <= len; i++) {
+            double d = (Math.random() * (122 - 97)) + 97;       // ASCII values between 97 to 122
+            int num = (int) d;
+            String s = Character.toString((char) num);
+            lc.append(s);
+        }
+        return lc.toString();
+    }
+
+    // Function to generate only uppercase letters of length specified in argument
+    public static String generateUCLetters(int len) {
+        StringBuilder uc = new StringBuilder();
+        for (int i = 1; i <= len; i++) {
```

Comment:
These for loops look remarkably similar except for the ASCII values. Can they be refactored into a helper method?

Reply:
They look similar, but the ASCII number generated for each one has to be in a different range and hence has to have an individual formula. Further, the characters are added iterations; hence, the ASCII formula needs to be inside a loop. This basic unit will look similar for all four functions and hasn't been merged due to the different ranges of the ASCII code generated.

Reply:
Oh, the helper could take in the two numbers defining the range. I think it is worth doing, since the change is easy to make and cuts down on duplicate code.